mutiny-core/src/lib.rs

```diff
@@ -2289,6 +2299,76 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and redeems/melts the token.
+    pub async fn melt_cashu_token(
+        &self,
+        token_v3: TokenV3,
+    ) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let mut invoices: Vec<MutinyInvoice> = Vec::with_capacity(token_v3.tokens.len());
+
+        for token in token_v3.tokens {
+            let mint_url = match token.mint {
+                Some(url) => url,
+                None => return Err(MutinyError::EmptyMintURLError),
+            };
+
+            let total_proofs_amount = token.proofs.total_amount();
+            // create invoice for 1% less than proofs amount
+            let mut invoice_amount = total_proofs_amount as f64 * (1.0 - 1.0 / 100.0);
+            let mut mutiny_invoice: MutinyInvoice;
+            let mut mutiny_invoice_str: Bolt11Invoice;
+            let mut melt_quote_res: PostMeltQuoteBolt11Response;
+
+            loop {
+                mutiny_invoice = self
+                    .create_invoice(invoice_amount as u64, vec![MELT_CASHU_TOKEN.to_string()])
+                    .await?;
+
+                mutiny_invoice_str = mutiny_invoice
+                    .bolt11
+                    .clone()
+                    .expect("The invoice should have BOLT11");
+
+                let quote_request = PostMeltQuoteBolt11Request {
+                    request: mutiny_invoice_str.to_string(),
+                    unit: CurrencyUnit::Sat,
+                };
+
+                melt_quote_res = self
+                    .cashu_client
+                    .mint_post(
+                        &mint_url.join("/v1/melt/quote/bolt11")?,
+                        json!(quote_request),
+                    )
```

Comment:
can make this a dedicated function on the cashu client. This way you can just pass the mint url and a `PostMeltQuoteBolt11Request`. Will make things easier to use in the future and we could eventually mock it for tests too